### ISSUE TITLE
PW Login fails sometimes

### ISSUE THREAD

**Issue by somatonic** [ryancramerdesign/ProcessWire#1254]
I think there were plenty of people having issues with PW login. Again a client of us can't login and she used the "enter" key and not login button.

It's not clear whats happening but definately something wrong.

I'm trying to doublicate and I can break the login easily in Chrome, FF etc...

1. Login with username and password, and press enter, login works
2. Logout and login again with entering a wrong password. Press enter and "login failed"
3. Now enter the correct password and press enter, you'll see "login failed" and the throttle time message.
4. Enter again the correct password and press button to login, again same login failed with throttle message.
5. It works again when waiting long enough 

This is same scenario for PW 2.4 until 2.6.1. 

Let me know if you can't reproduce and I'll cast a video.

**Comment by somatonic** [comment]
It seems I'm confused by throttle login function. Once there's a failed login the first time but still a throttle going, I don't see that message to wait x seconds but only the "login failed". Then after entering the correct password, it suddenly shows the throttle message.

**Comment by somatonic** [comment]
Looking at the co-worker there seems to definately a problem with using enter against pressing login button. This is in PW 2.4.

Login works fine when pressing button, but failes when pressing enter. It just shows 60 sec to wait since ever.

**Comment by plauclair** [comment]
Did you wait for the throttle to expire before sending again? This sounds like it works as expected. I tried it, didn't see anything weird on my part.

**Comment by somatonic** [comment]
Yes but if I remember in PW 2.4 there was a throttle expire problem. The maximum time (60s) always shows when something went wrong. Though the real weird thing is that it only happens when pressing enter. I disabled throttle module.

**Comment by plauclair** [comment]
I tried to replicate here, but can't. The timeout incrementally increases Either through the usual login page and a few custom ones through API like this one. Did you try with the database session module on and off?

```php
$erreur = null;
$pageContent = null;

// Logout function
if (isset($input->get->logout)) {
	$session->logout();
	$session->redirect($config->urls->root);
}

// Login with redirect
if ($input->post->submit) {
	$username = $sanitizer->username($input->post->username);
	$password = $sanitizer->text($input->post->password);

	try {
		$userLogin = $session->login($username,$password);
		if ($userLogin) {
			$pageId = $sanitizer->text($input->post->page);
			$session->redirect($pages->get("id={$pageId}")->httpUrl);
		} else {
			$erreur = "Échec de la connexion";
		}
	} catch(Exception $e) {
		$erreur = $e->getMessage();
	}
}


// Login form
if ($user->isLoggedin()) {
	$pageContent = "Vous êtes déjà connecté";
} else {
	if (isset($erreur)) { $pageContent .= "<div class='erreur'>$erreur</div>"; }
	$pageContent .= '<form action="" method="post" class="form">';
	$pageContent .= '<label for="username">Nom d\'utilisateur</label>';
	$pageContent .= '<input type="text" name="username"/>';
	$pageContent .= '<label for="password">Mot de passe</label>';
	$pageContent .= '<input type="password" name="password"/>';
	$pageContent .= '<input type="hidden" name="page" value="' . $input->get->page . '""/>';
	$pageContent .= '<input type="submit" name="submit" value="Se connecter"/>';
	$pageContent .= '</form>';
}
```

**Comment by somatonic** [comment]
It's the admin login. And in PW 2.4 there were no DB Session Handler.

**Comment by somatonic** [comment]
Here's a screencast. 

1. Login first time with typing password and keyboard enter works
2. Logout
3. Login again with typing in the password fails and shows 60s message

https://www.youtube.com/watch?v=IoTbgQ-0rHM

**Comment by ryancramerdesign** [comment]
I'm aware there was an issue with this back in 2.4, but it was fixed in 2.5 (I think with your help if I recall correctly). At least I can't duplicate the issue on current master or dev version of PW. Let me know if you are seeing the issue with 2.6.

**Comment by isellsoap** [comment]
@somatonic This issue doesn’t seem to be reproducible anymore, thus I close it. If you think the issue requires additional work to do, please consider reopening it.
